lib/rules/no-underscore-dangle.js

```diff
@@ -145,6 +174,7 @@ module.exports = {
                     });
                 }
             }
+            checkForFollowingUnderscoreInFunctionParameter(node);
```

Comment:
Can we combine `checkForFollowingUnderscoreInFunctionParameter` and `checkForTrailingUnderscoreInFunctionDeclaration` into one method and use it to validate `FunctionDeclaration`, `FunctionExpression`, and `ArrowFunctionExpression` nodes? 

It looks like we can copy the body of `checkForFollowingUnderscoreInFunctionParameter`  directly into this function, since it already has the `if (node.id)` guard. 

This would also allow us to validate function names for named function expressions (`const hello = function _hello() {};`), which I don't think are currently covered.